tests/test_utils.py

```diff
@@ -237,3 +238,88 @@ def test_pad_across_processes(self):
         with self.assertWarns(CannotPadNestedTensorWarning):
             nt2 = pad_across_processes(nt)
         self.assertIs(nt, nt2)
+
+    def test_slice_and_concatenate(self):
+        # Should be equivalent to the slice func used in `pippy_forward`
```

Comment:
Also it would be great to have a pippy example that was failing before this PR (2 gpus and a `batch_size` of 1 for example)

Reply:
Done :)